nuget/lib/dependabot/nuget/update_checker/repository_finder.rb

```diff
@@ -307,6 +310,24 @@ def add_config_file_credentials(sources:, doc:)
         end
         # rubocop:enable Metrics/PerceivedComplexity
 
+        def expand_windows_style_environment_variables(string)
+          # NuGet.Config files can have Windows-style environment variables that need to be replaced
+          # https://learn.microsoft.com/en-us/nuget/reference/nuget-config-file#using-environment-variables
+          string.gsub(/%([^%]+)%/) do
+            environment_variable_name = ::Regexp.last_match(1)
+            environment_variable_value = ENV.fetch(T.must(environment_variable_name), nil)
```

Comment:
Move `T.must()` up to the caller, otherwise the type is `T.nilable(String)` everywhere else

```suggestion
            environment_variable_name = T.must(::Regexp.last_match(1))
            environment_variable_value = ENV.fetch(environment_variable_name, nil)
```